packages/calypso-e2e/src/lib/blocks/block-flows/shared/block-helpers.ts

```diff
@@ -44,10 +44,29 @@ export async function validatePublishedFormFields(
  * @param {Locator} parentFormBlock Locator for the parent form block.
  * @param {FieldLabelDetails} details The details for labeling.
  */
-export async function labelFormFieldBlock( parentFormBlock: Locator, details: FieldLabelDetails ) {
-	const { blockName, accessibleLabelName, labelText } = details;
-	await parentFormBlock
+export async function labelFormFieldBlock(
+	parentFormBlock: Locator,
+	{
+		blockName,
+		accessibleLabelName,
+		labelText,
+		parentBlockName = undefined,
+		isRefactor = false,
+	}: {
+		blockName: string;
+		accessibleLabelName: string;
+		labelText: string;
+		parentBlockName?: string;
+		isRefactor?: boolean;
+	}
+) {
+	let scope = parentFormBlock;
+	if ( isRefactor && parentBlockName ) {
+		scope = scope.locator( makeSelectorFromBlockName( parentBlockName ) );
+	}
```

Comment:
In the refactored form fields, we re-use the same 'Label' and 'Input' and 'Option' blocks within multiple fields. To ensure we can find and label those common blocks correctly, we need to pass in the parent block (ie, we add one label to Options within Radio blocks, and another to Options with Checklist blocks).